tests/test_http.py

```diff
@@ -26,6 +27,11 @@ def s():
 
 def has_connection():
     # In case we run tests in an environment without internet connection.
+
+    if sys.version_info < (2, 7):
+        # py 2.6 doesn't have SNI support, required for httpbin, as it has SSLv3 certificates
+        return False
```

Comment:
Its related to http://docs.python-requests.org/en/master/community/faq/#what-are-hostname-doesn-t-match-errors:

>  If you're certain the server's SSL setup is correct (for example, because you can visit the site with your browser) and you're using Python 2.6 or 2.7, a possible explanation is that you need Server-Name-Indication.

>Server-Name-Indication, or SNI, is an official extension to SSL where the client tells the server what hostname it is contacting. This is important when servers are using Virtual Hosting. When such servers are hosting more than one SSL site they need to be able to return the appropriate certificate based on the hostname the client is connecting to.

>Python3 and Python 2.7.9+ include native support for SNI in their SSL modules

py2.6 has no native SNI support and although there seems to be a way to make it work in py2.6 I'm not sure how to do that :(